cmd/kube-apiserver/app/aggregator.go

```diff
@@ -205,6 +205,7 @@ var apiVersionPriorities = map[schema.GroupVersion]priority{
 	// to my knowledge, nothing below here collides
 	{Group: "apps", Version: "v1beta1"}:                          {group: 17800, version: 1},
 	{Group: "apps", Version: "v1beta2"}:                          {group: 17800, version: 1},
```

Comment:
should these have equal priority?

Reply:
It may be due to different API groups have different priorities. `extensions/v1beta1` daemonset works fine and passes the test, but not `apps/v1beta1` statefulset. `apps/v1` statefulset is preferred by default, which fails the 2 kubectl tests.

Reply:
To answer your question, I am not sure about it. I lean towards having them with different priorities, but I didn't work on apiserver/aggregator codes before.

Reply:
I tried to ask how priority should be set but didn't get a response yet https://github.com/kubernetes/kubernetes/issues/49391#issuecomment-332912633